### PR TITLE
New move code: Only snap to collision box only on fall

### LINKED ISSUE #11261: Sneaking while jump causes teleporting
##### Minetest version
```
5.5.0-dev
```
##### Summary
If you jump and move into another node, being in air, pressing Shift teleports you to the edge of node, even if it is far from you.
Not work if node above is not air, so it seems obvious that this is a consequence of "use sneak to not to fall from edge" mechanic.

##### Steps to reproduce
Build a slab (better 1/16) tilted 90° and jump towards it from non-occupied side, pressing Shift

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #11261

I did not mark this as trivial because there might (as always) someone depend on this snapping behaviour already when jumping up.

## To do

This PR is Ready for Review.

## How to test

1. Testing mod:

```Lua
minetest.register_on_joinplayer(function(player)
	player:set_physics_override({
		sneak_glitch = true,
		sneak = true,
		new_move = true,
		jump = 2,
	})
end)
```

2. Stack a few slabs (like below) and rotate them accordingly to make the effect more apparent:

   * https://github.com/minetest/minetest/issues/11261#issuecomment-840746330

### CODE REVIEW

**Comment by SmallJoker** [comment]
@NeroBurner @RichardTry  Does this work as you'd expect?

**Comment by NeroBurner** [comment]
It works half of the time. During the jump while going up it works as expected. After reaching the top and starting falling down the "teleport" comes into effect

**Comment by SmallJoker** [comment]
Yes, that seems entirely as intended so that you could adhere to nearby nodes to prevent falling down. Modifying that would imply a breaking change to the gameplay.

**Comment by RichardTry** [comment]
The correct behavior seems to be like this: Sneak prevents you from falling just if you touch ground (edge), sneak should not prevent you from falling if you "fly" over the edge.
P.S. I can't compile this PR with up-to-date IrrlichtMt

**Comment by NeroBurner** [comment]
@RichardTry yes I think that's what I would expect

building works for me using IrrlichtMt as a subdirectory (also when rebasing the PR commit on current master)

```sh
git clone --depth 1 https://github.com/minetest/irrlicht.git lib/irrlichtmt
```

**Comment by RichardTry** [comment]
I found out that sneaking "teleports" me to the top of node, no matter which collision box it has. This should be the reason of a problem.
I'll try to compile this PR.

**Comment by SmallJoker** [comment]
To test this PR, you'll either need an Irrlicht mt1 clone or you could apply the patches to current master. [example script](https://gist.github.com/SmallJoker/4d411b958295d9e694cd9aa5984563db).

**Comment by erlehmann** [comment]
I was not able to find a way where this actually changes gameplay while trying for a few minutes.

I'd be wary of changinng this though – if anything relies on it, it is most likely parkour maps, where this change may turn hard jumps into impossible jumps without anyone noticing.

**Comment by erlehmann** [comment]
I *think* I figured it out: Snapping when jumping up is *probably* a part of the “sneak jump” feature, which allows players to jump/teleport on a node that is normally too high to jump on if they press sneak, therefore having a way to “jump a bit more”. This is mainly used with fences and walls.

**TL;DR: Please do not merge this.**

**Comment by Zughy** [comment]
@SmallJoker the issue you had mentioned in the very beginning it's been closed as "Not a bug" + Erlehmann comment. Should I close this?

**Comment by SmallJoker** [comment]
@Zughy Whereas it might not be a bug, I consider this as a movement improvement. Snapping is only relevant to get onto nodes while falling, hence it should only happen in that case.

**Comment by SmallJoker** [comment]
Wrong implementation idea. Holding sneak acts like grabbing to the top surface of a node, hence snapping makes sense for upwards movements. If even, it should be the opposite: no snapping when falling down because the body's center of mass is too far away from the edge to have a chance to grab the node.
